webapp/src/Controller/Team/SubmissionController.php

```diff
@@ -201,4 +207,45 @@ public function viewAction(Request $request, int $submitId)
             return $this->render('team/submission.html.twig', $data);
         }
     }
+
+    /**
+     * @Route("/submission/{submitId<\d+>}/download", name="team_submission_download")
+     * @param int $submitId
+     *
+     * @return Response
+     * @throws NonUniqueResultException
+     */
+    public function downloadAction($submitId)
+    {
+        $allowDownload = (bool)$this->config->get('allow_team_submission_download');
+        if (!$allowDownload) {
+            throw new NotFoundHttpException('Submission download not allowed');
+        }
+
+        $user = $this->dj->getUser();
+        $team = $user->getTeam();
+        /** @var Submission $submission */
+        $submission = $this->em->createQueryBuilder()
+            ->from(Submission::class, 's')
+            ->join('s.judgings', 'j')
+            ->join('s.contest_problem', 'cp')
+            ->join('cp.problem', 'p')
+            ->join('s.language', 'l')
+            ->join('s.files', 'f')
+            ->select('s, f')
+            ->andWhere('j.submitid = :submitId')
+            ->andWhere('j.valid = 1')
```

Comment:
why do we need a valid judging for this?

Reply:
We don’t but we show it in the pop-up that does. I can remove this here, then you can access it if you know the url and later possibly update the UI

Reply:
Indeed, so also my comment above: I see no need to restrict this to anything more than the right team and submission.

Reply:
Agreed